fmt: bad INTERFACE_INCLUDE_DIRECTORIES in cmake targets for fmt

**Describe the bug**

fmt has two derivations, out and dev. The cmake file ends up with a path that contains the incorrect/missing path to the 'out' derivations 'include' folder. It should be the 'dev' folder'.

The eventual error you get when you end up with package A depending on package B and package B  depending on fmt is:

```
CMake Error in yarpl/CMakeLists.txt:
  Imported target "Folly::folly" includes non-existent path

    "/nix/store/bx1ksiw20qmqmkbbylrxdabfmipk49x7-fmt-6.2.0/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

when the INTERFACE_INCLUDE_DIRECTORIES should be /nix/store/...-fmt-6.2.0-dev/include`. 

**To Reproduce**
Steps to reproduce the behavior:
1. build fmt package
2. check cmake files for fmt path

```
default 000 > nix repl '<nixpkgs/nixos>'
Welcome to Nix version 2.3.3. Type :? for help.

Loading '<nixpkgs/nixos>'...
Added 6 variables.

#extras/nix/fmt.nix is just a copy of the upstream master 6.3.0 fmt derivation
nix-repl>   fmt = (pkgs.callPackage ./extras/nix/fmt.nix {})

nix-repl> fmt.postInstall
error: attribute 'postInstall' missing, at (string):1:1

nix-repl> :b fmt

default 000 > grep IMPORT_PREFIX /nix/store/4azmpr6bbiz130sj4nyr3gq79ak381vb-fmt-6.2.0-dev/l
ib/cmake/fmt/fmt-targets.cmake
set(_IMPORT_PREFIX "/nix/store/bx1ksiw20qmqmkbbylrxdabfmipk49x7-fmt-6.2.0")
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
set(_IMPORT_PREFIX)

default 000 > stat /nix/store/bx1ksiw20qmqmkbbylrxdabfmipk49x7-fmt-6.2.0/include
stat: cannot stat '/nix/store/bx1ksiw20qmqmkbbylrxdabfmipk49x7-fmt-6.2.0/include': No such file or directory
```

**Expected behavior**

fmt-targets.cmake  should point to fmt-dev....:

```
default 000 > nix repl '<nixpkgs/nixos>'
Welcome to Nix version 2.3.3. Type :? for help.

Loading '<nixpkgs/nixos>'...
Added 6 variables.

nix-repl>   fmt = (pkgs.callPackage ./extras/nix/fmt.nix {})

nix-repl> :b fmt
[1 built, 0.0 MiB DL]

this derivation produced the following outputs:
  dev -> /nix/store/22j1xg7jahq28071cgl6vwdsg4hhymk8-fmt-6.2.0-dev
  out -> /nix/store/6z2lnhc1n76iaf72a5gkpdmvm8ajb1y6-fmt-6.2.0

nix-repl> fmt.postInstall
"substituteInPlace \"$out\"/lib/cmake/fmt/fmt-targets.cmake \\\n  --replace 'INTERFACE_INCLUDE_DIRECTORIES \"${_IMPORT_PREFIX}/include\"' 'INTERFACE_INCLUDE_DIRECTORIES \"'$dev'\"'\n"

nix-repl>

default 000 > grep INCLUDE_DIR /nix/store/22j1xg7jahq28071cgl6vwdsg4hhymk8-fmt-6.2.0-dev/lib
/cmake/fmt/fmt-targets.cmake
  INTERFACE_INCLUDE_DIRECTORIES "/nix/store/22j1xg7jahq28071cgl6vwdsg4hhymk8-fmt-6.2.0-dev"
  INTERFACE_INCLUDE_DIRECTORIES "/nix/store/22j1xg7jahq28071cgl6vwdsg4hhymk8-fmt-6.2.0-dev"
```

**Additional context**

Looks like the bug is resolved https://github.com/NixOS/nixpkgs/pull/63097 but is stalled since 2019. That stack of diffs is exactly what I was doing when i ran into this bug, building fbthrift+rsocket-cpp. and the solution i came up with locally to get 'expected behavior' is the same as well.

**Notify maintainers**

@jdehaas

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.113, NixOS, 19.09.2477.a7ceb2536ab (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.3`
 - channels(sheenobu): `"home-manager"`
 - channels(root): `"nixos-19.09.2477.a7ceb2536ab"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/libraries/fmt/default.nix
+++ b/pkgs/development/libraries/fmt/default.nix
@@ -2,7 +2,7 @@
 
 stdenv.mkDerivation rec {
   pname = "fmt";
-  version = "6.2.0";
+  version = "6.2.1";
 
   outputs = [ "out" "dev" ];
 
@@ -10,7 +10,7 @@ stdenv.mkDerivation rec {
     owner = "fmtlib";
     repo = "fmt";
     rev = version;
-    sha256 = "1vr08a8q94j66gas7ddbpdbq72b2ikd0mkgd5zd3l63610n8qajs";
+    sha256 = "1i6nfxazq4d05r3sxyc3ziwkqq7s8rdbv9p16afv66aqmsbqqqic";
   };
 
   patches = [
@@ -27,6 +27,12 @@ stdenv.mkDerivation rec {
     (fetchpatch {
       url = "https://github.com/fmtlib/fmt/commit/78f041ab5b40a1145ba686aeb8013e8788b08cd2.patch";
       sha256 = "hjE6Q/ubA4UhvuJXgcsA3wiGoDK031P19njQRL9JF8M=";
+    })
+
+    # Fix cmake config paths.
+    (fetchpatch {
+      url = "https://github.com/fmtlib/fmt/pull/1702.patch";
+      sha256 = "18cadqi7nac37ymaz3ykxjqs46rvki396g6qkqwp4k00cmic23y3";
     })
   ];
 

